--- install/index.php
+++ install/index.php
@@ -508,15 +508,16 @@
 		if(!File_Exists($File)){
 			#-------------------------------------------------------------------------------
 			Array_UnShift($HostsIDs,HOST_ID);
 			#-------------------------------------------------------------------------------
-			if(!File_Put_Contents($File,SPrintF("HostsIDs=%s\nmemcache.port=11211",Implode(',',$HostsIDs))))
-				Error(SPrintF('Ошибка записи файла (%s)',$File));
-			#-------------------------------------------------------------------------------
-			if(!File_Put_Contents($File,SPrintF("HostsIDs=%s\nmemcached.port=11211",Implode(',',$HostsIDs))))
-				Error(SPrintF('Ошибка записи файла (%s)',$File));
-			#-------------------------------------------------------------------------------
-			if(!File_Put_Contents($File,SPrintF("HostsIDs=%s\nredis.port=6379",Implode(',',$HostsIDs))))
+			$CacheConfig = SPrintF("HostsIDs=%s",Implode(',',$HostsIDs));
+			if(Extension_Loaded('memcache'))
+				$CacheConfig .= "\nmemcache.port=11211";
+			if(Extension_Loaded('memcached'))
+				$CacheConfig .= "\nmemcached.port=11211";
+			if(Extension_Loaded('redis'))
+				$CacheConfig .= "\nredis.port=6379";
+			if(!File_Put_Contents($File,$CacheConfig))
 				Error(SPrintF('Ошибка записи файла (%s)',$File));
 			#-------------------------------------------------------------------------------
 		}
 		#-------------------------------------------------------------------------------
