--- install/index.php
+++ install/index.php
@@ -527,7 +527,7 @@
 			if(!@MkDir($Link,0755,TRUE))
 				Error(SPrintF('Не возможно создать директорию (%s)',$Link));
 			#-------------------------------------------------------------------------------
-			if(!@SymLink(SPrintF('./hosts/%s/tmp/public',HOST_ID),'./public'))
+			if(!@SymLink(SPrintF('%s/hosts/%s/tmp/public', SYSTEM_PATH, HOST_ID), SPrintF('%s/public', SYSTEM_PATH)))
 				Error(SPrintF('Не возможно создать символическую ссылку (%s)',$Link));
 			#-------------------------------------------------------------------------------
 		}
